test/highlight/types.cs

```diff
@@ -0,0 +1,47 @@
+using System.Runtime.Intrinsics.X86;
+
+class A
+{
+    public void M()
+    {
+        int a;
+        // <- type.builtin
+        var a;
+        // <- type.builtin
+
+        int? a;
+        // <- type.builtin
+        // ^ operator
+        A? a;
+        // <- type
+         // <- operator
+
+        int* a;
+        // <- type.builtin
+        // ^ operator
+        A* a;
+        // <- type
+         // <- operator
+
+        ref A* a;
+        // <- keyword
+        //  ^ type
+        //   ^ operator
+
+        var a = x is int;
+        //           ^ type.builtin
+        var a = x is A;
+        //           ^
```

Comment:
This is not highlighted due to https://github.com/tree-sitter/tree-sitter-c-sharp/issues/317.